src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingParameterResolver.cs

```diff
@@ -0,0 +1,123 @@
+        }
+
+        public abstract string Name { get; }
+
+        public static bool IsSystemParameter(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                throw new ArgumentNullException("value");
+            }
+
+            BindingParameterResolver resolver = null;
+            return TryGetResolver(value, out resolver);
+        }
+
+        public static bool TryGetResolver(string value, out BindingParameterResolver resolver)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                throw new ArgumentNullException("value");
+            }
+
+            resolver = _resolvers.FirstOrDefault(p => value.StartsWith(p.Name, StringComparison.OrdinalIgnoreCase));
+            return resolver != null;
+        }
+
+        public abstract string Resolve(string value);
+
+        protected string GetFormatOrNull(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                throw new ArgumentNullException("value");
+            }
+
+            if (!value.StartsWith(Name, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture, "The value specified is not a '{0}' binding parameter.", Name), "value");
+            }
+
+            if (value.Length > Name.Length && value[Name.Length] == ':')
+            {
+                // we have a value of format <Name>:<Format>
+                // parse out everything after the first colon
+                int idx = Name.Length;
+                return value.Substring(idx + 1);
+            }
+
+            return null;
+        }
+
+        private class RandGuidResolver : BindingParameterResolver
+        {
+            public override string Name
+            {
+                get
+                {
+                    return "rand-guid";
+                }
+            }
+
+            public override string Resolve(string value)
+            {
+                string format = GetFormatOrNull(value);
+
+                if (!string.IsNullOrEmpty(format))
+                {
+                    return Guid.NewGuid().ToString(format, CultureInfo.InvariantCulture);
+                }
+                else
+                {
+                    return Guid.NewGuid().ToString();
+                }
+            }
+        }
+
+        private class DateTimeResolver : BindingParameterResolver
+        {
+            public override string Name
+            {
+                get
+                {
+                    return "datetime";
+                }
+            }
+
+            public override string Resolve(string value)
+            {
+                string format = GetFormatOrNull(value);
+
+                if (!string.IsNullOrEmpty(format))
+                {
+                    return DateTime.Now.ToString(format, CultureInfo.InvariantCulture);
```

Comment:
Using local rather than utc so that users can control this via the WEBSITE_TIME_ZONE WebApp setting. By default, VMs will already be using UTC, but if people want to change their timezone they can via this setting, and using Local time will allow that to be picked up properly. See https://github.com/Azure/azure-webjobs-sdk-extensions/issues/32 for details.